stream-access-control/access-control-core/src/main/java/com/backbase/stream/service/UserService.java

```diff
@@ -273,30 +275,48 @@ public Mono<LegalEntity> linkLegalEntityToRealm(LegalEntity legalEntity) {
      * @return the same User with updated internal and external id on success
      */
     public Mono<User> createOrImportIdentityUser(User user, String legalEntityInternalId, StreamTask streamTask) {
-        CreateIdentityRequest createIdentityRequest = new CreateIdentityRequest();
-        createIdentityRequest.setLegalEntityInternalId(legalEntityInternalId);
-        createIdentityRequest.setExternalId(user.getExternalId());
-        createIdentityRequest.setAdditions(user.getAdditions());
+
+        Mono<CreateIdentityResponse> upsertCall;
 
         if (IdentityUserLinkStrategy.CREATE_IN_IDENTITY.equals(user.getIdentityLinkStrategy())) {
             Objects.requireNonNull(user.getFullName(), "User Full Name is required for user: " + user.getExternalId() + " in legal entity: " + legalEntityInternalId);
             Objects.requireNonNull(user.getEmailAddress(), "User Email Address is required for user: " + user.getExternalId() + " in legal entity: " + legalEntityInternalId);
             Objects.requireNonNull(user.getMobileNumber(), "User Mobile Number is required for user: " + user.getExternalId() + " in legal entity: " + legalEntityInternalId);
 
+            CreateIdentityRequest createIdentityRequest = new CreateIdentityRequest();
+            createIdentityRequest.setLegalEntityInternalId(legalEntityInternalId);
+            createIdentityRequest.setExternalId(user.getExternalId());
+            createIdentityRequest.setAdditions(user.getAdditions());
             createIdentityRequest.setFullName(user.getFullName());
             createIdentityRequest.setEmailAddress(user.getEmailAddress().getAddress());
```

Comment:
The user email is a required field in DBS when using CREATE_IN_IDENTITY, and it is very hard to identify in the DBS logs when the value here is null, could we include a validation here so we could fail fast with a clear log indication that this is a required field?

Reply:
Hi @uesleilima , it is there, I didn't remove it:
`        if (IdentityUserLinkStrategy.CREATE_IN_IDENTITY.equals(user.getIdentityLinkStrategy())) {
            Objects.requireNonNull(user.getFullName(), "User Full Name is required for user: " + user.getExternalId() + " in legal entity: " + legalEntityInternalId);
            Objects.requireNonNull(user.getEmailAddress(), "User Email Address is required for user: " + user.getExternalId() + " in legal entity: " + legalEntityInternalId);
            Objects.requireNonNull(user.getMobileNumber(), "User Mobile Number is required for user: " + user.getExternalId() + " in legal entity: " + legalEntityInternalId);

            CreateIdentityRequest createIdentityRequest = new CreateIdentityRequest();
            createIdentityRequest.setLegalEntityInternalId(legalEntityInternalId);
            createIdentityRequest.setExternalId(user.getExternalId());
            createIdentityRequest.setAdditions(user.getAdditions());
            createIdentityRequest.setFullName(user.getFullName());
            createIdentityRequest.setEmailAddress(user.getEmailAddress().getAddress());
            createIdentityRequest.setMobileNumber(user.getMobileNumber().getNumber());
            ofNullable(user.getAttributes()).ifPresent(createIdentityRequest::setAttributes);`

Reply:
Unfortunately, only checking the wrapper object doesn't work; let's improve it by checking the underlying string property. e.g. the `.getAddress()`:
```
Objects.requireNonNull(user.getEmailAddress())**.getAddress(),** "User Email Address is required for user: " + user.getExternalId() + " in legal entity: " + legalEntityInternalId);
Objects.requireNonNull(user. getMobileNumber()**. getNumber()**, "User Mobile Number is required for user: " + user.getExternalId() + " in legal entity: " + legalEntityInternalId);
```

Reply:
@uesleilima Yep I see it now. Fixed. Added a test.